web/app/tournament/tourny_helpers.py

```diff
@@ -0,0 +1,79 @@
+from flask import current_app, render_template, redirect, url_for
+from flask import jsonify, request, Response, flash
+from app.models import db, Tournament, TournamentPlayer, Match, Round
+
+def empty_field(form_data, fetched_db_data):
+    """Treating an empty string returned by form as equal to None in DB"""
+    if (form_data == None and fetched_db_data == '') or (form_data == '' and fetched_db_data == None):
+        return True
+    else:
+        return False
+
+
+def match_edit(form, match, tournament_id):
```

Comment:
This is a good first step, but what I was thinking was, in particular, to break out tho logic from the database (model) and the form (the view). 

So, I was imagining that

```python
# This method doesn't talk to the database or deal with forms or urls. It's pure logic.
def match_maker(players, match_options):
  # Do the match
  # Return the matches

You could still have the match_edit method:

```python
def match_edit(form, match, tournament_id):
  all_players = get_all_players(tournament_id)
  match_options = get_make_options(form, tournament_id)
  current_matches = get_matches(tournament_id)
  new_matches = match_maker(all_players, match_options)
  store_new_matches(new_matches)
  perform_redirect(tournament_id, new_matches)
```